pkg/controllers/restore/controller.go

```diff
@@ -477,6 +474,31 @@ func (h *handler) generateDependencyGraph(ownerToDependentsList map[string][]res
 	return nil
 }
 
+// customize provides customization of restored resource for edge cases
+func customize(obj *unstructured.Unstructured) {
+	switch obj.GetKind() {
+	case "ServiceAccount":
+		// remove secrets section as referenced secrets will be removed by k8s Token Controller as they are considered orphaned
+		delete(obj.Object, secretsMapKey)
+		logrus.Debugf("Secrets section is ignored in ServiceAccount %s/%s", obj.GetNamespace(), obj.GetName())
+	case "Cluster":
+		// for fleet cluster it needs to be reimported in order to reissue service account token that is no longer valid
+		if obj.GetAPIVersion() == "fleet.cattle.io/v1alpha1" {
+			// only warn error if field can't be found. In case there are breaking api changes error will be printed as Warn and user has workaround to patch it.
+			redeployAgentGeneration, ok, err := unstructured.NestedInt64(obj.Object, "spec", "redeployAgentGeneration")
+			if err != nil {
+				logrus.Warnf("Fleet cluster %s/%s can't be reset to be re-imported, failed to fetch .spec.redeployAgentGeneration", obj.GetNamespace(), obj.GetName())
```

Comment:
Recommend adding `err` message to the message we're logging as otherwise we don't know why it happened. Same for the other WARN.